many-fuzzy/src/main.rs

```diff
@@ -0,0 +1,348 @@
+use crate::stats::Statistics;
+use clap::Parser;
+use fuzz::FuzzGenerator;
+use many::message::{
+    decode_response_from_cose_sign1, encode_cose_sign1_from_request, RequestMessage,
+    RequestMessageBuilder,
+};
+use many::types::identity::CoseKeyIdentity;
+use many::Identity;
+use minicose::CoseSign1;
+use rand::{Rng, SeedableRng};
+use tokio::sync::oneshot::Sender;
+use tokio::task::JoinHandle;
+use tokio::time::Instant;
+use tokio::time::{interval, Duration};
+use tracing::level_filters::LevelFilter;
+
+mod fuzz;
+mod parsers;
+mod stats;
+
+#[derive(Parser)]
+struct Opts {
+    /// Increase output logging verbosity to DEBUG level.
+    #[clap(short, long, parse(from_occurrences))]
+    verbose: i8,
+
+    /// Suppress all output logging. Can be used multiple times to suppress more.
+    #[clap(short, long, parse(from_occurrences))]
+    quiet: i8,
+
+    #[clap(subcommand)]
+    subcommand: SubCommand,
+}
+
+#[derive(Parser)]
+enum SubCommand {
+    /// Run the fuzzer against a server.
+    Fuzz(FuzzOpt),
+}
+
+#[derive(Parser)]
+struct FuzzOpt {
+    /// Number of messages to create and send.
+    #[clap(short)]
+    number: u64,
+
+    /// A seed to use for the Rng. By default does not seed the Rng and instead
+    /// randomize it on start.
+    #[clap(long)]
+    seed: Option<u64>,
+
+    /// One or more pem file(s) to sign messages, separated by a comma.
+    /// If this is omitted, the message will be anonymous.
+    /// The special string "anonymous" will be used to add anonymous
+    /// to the list of pem files.
+    #[clap(long)]
+    pem: Option<String>,
+
+    /// The identity to send it to.
+    #[clap(long)]
+    to: Option<Identity>,
+
+    /// The server to connect to.
+    server: url::Url,
+
+    /// The endpoint to call.
+    endpoint: String,
+
+    /// The content of the message itself (its payload).
+    /// Will replace any substring `%()` with fuzzed values.
+    data: String,
+}
+
+fn split_pem_args(pem_list: String) -> Result<Vec<CoseKeyIdentity>, String> {
+    pem_list
+        .split(",")
```

Comment:
Done. It does, but it needs to specify the flag again, so it would be `--pem ~/id1.pem --pem ~/id2.pem ...` which I don't like. I'll accept both `,` and split.